File: Algorithm/Go/1000+/1207. Unique Number of Occurrences.go
func uniqueOccurrences(arr []int) bool {
	counter := make(map[int]int)
	for _, num := range arr {
		counter[num]++
	}
	unique := make(map[int]bool)
	for _, v := range counter {
		if unique[v] {
			return false
		}
		unique[v] = true
	}
	return true
}

File: Algorithm/C++/1000+/1207. Unique Number of Occurrences.cc
#include <unordered_map>
#include <unordered_set>
#include <vector>

class Solution {
public:
    bool uniqueOccurrences(std::vector<int>& arr) {
        std::unordered_map<int, int> count;
        for (auto num: arr) {
          count[num]++;
        }
        std::unordered_set<int> unique;
        for (auto e : count) {
          unique.insert(e.second);
        }
        return count.size() == unique.size();
    }
};